File: phenomenal/ifs_100.rb
class Foo
  def test(mode)
    if mode==1 
      1
    elsif mode==2 
      2
    elsif mode==3 
      3
    elsif mode==4 
      4
    elsif mode==5 
      5
    elsif mode==6 
      6
    elsif mode==7 
      7
    elsif mode==8 
      8
    elsif mode==9 
      9
    elsif mode==10 
      10
    elsif mode==11 
      11
    elsif mode==12 
      12
    elsif mode==13 
      13
    elsif mode==14 
      14
    elsif mode==15 
      15
    elsif mode==16 
      16
    elsif mode==17 
      17
    elsif mode==18 
      18
    elsif mode==19 
      19
    elsif mode==20 
      20
    elsif mode==21 
      21
    elsif mode==22 
      22
    elsif mode==23 
      23
    elsif mode==24 
      24
    elsif mode==25 
      25
    elsif mode==26 
      26
    elsif mode==27 
      27
    elsif mode==28 
      28
    elsif mode==29 
      29
    elsif mode==30 
      30
    elsif mode==31 
      31
    elsif mode==32 
      32
    elsif mode==33 
      33
    elsif mode==34 
      34
    elsif mode==35 
      35
    elsif mode==36 
      36
    elsif mode==37 
      37
    elsif mode==38 
      38
    elsif mode==39 
      39
    elsif mode==40 
      40
    elsif mode==41 
      41
    elsif mode==42 
      42
    elsif mode==43 
      43
    elsif mode==44 
      44
    elsif mode==45 
      45
    elsif mode==46 
      46
    elsif mode==47 
      47
    elsif mode==48 
      48
    elsif mode==49 
      49
    elsif mode==50 
      50
    elsif mode==51 
      51
    elsif mode==52 
      52
    elsif mode==53 
      53
    elsif mode==54 
      54
    elsif mode==55 
      55
    elsif mode==56 
      56
    elsif mode==57 
      57
    elsif mode==58 
      58
    elsif mode==59 
      59
    elsif mode==60 
      60
    elsif mode==61 
      61
    elsif mode==62 
      62
    elsif mode==63 
      63
    elsif mode==64 
      64
    elsif mode==65 
      65
    elsif mode==66 
      66
    elsif mode==67 
      67
    elsif mode==68 
      68
    elsif mode==69 
      69
    elsif mode==70 
      70
    elsif mode==71 
      71
    elsif mode==72 
      72
    elsif mode==73 
      73
    elsif mode==74 
      74
    elsif mode==75 
      75
    elsif mode==76 
      76
    elsif mode==77 
      77
    elsif mode==78 
      78
    elsif mode==79 
      79
    elsif mode==80 
      80
    elsif mode==81 
      81
    elsif mode==82 
      82
    elsif mode==83 
      83
    elsif mode==84 
      84
    elsif mode==85 
      85
    elsif mode==86 
      86
    elsif mode==87 
      87
    elsif mode==88 
      88
    elsif mode==89 
      89
    elsif mode==90 
      90
    elsif mode==91 
      91
    elsif mode==92 
      92
    elsif mode==93 
      93
    elsif mode==94 
      94
    elsif mode==95 
      95
    elsif mode==96 
      96
    elsif mode==97 
      97
    elsif mode==98 
      98
    elsif mode==99 
      99
    elsif mode==100 
      100
    end 
   end 
end

def testContexts
  $i = 0
  $ITER = 1000
  $CALLS = 100
  beginning_time = Time.now
  foo = Foo.new
  until $i > $ITER do
    num = 1 + rand(100)
    j = 0
    for j in (1..$CALLS)
      foo.test(num)
    end
    $i +=1;
  end
  return (Time.now - beginning_time)*1000
end

def runs(run)
  $times = 0
  File.open("runs_ifs/100/test#{run}.csv", 'w') {|f| f.write("iteration, Elapsed time in milliseconds
")}
  for $times in 1..100
    elapsed_time = testContexts
    File.open("runs_ifs/100/test#{run}.csv", 'a') do |f|
        f.puts("#{$times}, #{elapsed_time}")
      end
  end
end

for run in 1..10
  runs(run)
end

File: ContextTraits/ifs/ifs_100.js
var Foo = function() {};
Foo.prototype.test = function(mode) {
    if(mode==1) { 
    return  1;
    }
    else if(mode==2) {
        return 2;
    }
    else if(mode==3) {
        return 3;
    }
    else if(mode==4) {
        return 4;
    }
    else if(mode==5) {
        return 5;
    }
    else if(mode==6) {
        return 6;
    }
    else if(mode==7) {
        return 7;
    }
    else if(mode==8) {
        return 8;
    }
    else if(mode==9) {
        return 9;
    }
    else if(mode==10) {
        return 10;
    }
    else if(mode==11) {
        return 11;
    }
    else if(mode==12) {
        return 12;
    }
    else if(mode==13) {
        return 13;
    }
    else if(mode==14) {
        return 14;
    }
    else if(mode==15) {
        return 15;
    }
    else if(mode==16) {
        return 16;
    }
    else if(mode==17) {
        return 17;
    }
    else if(mode==18) {
        return 18;
    }
    else if(mode==19) {
        return 19;
    }
    else if(mode==20) {
        return 20;
    }
    else if(mode==21) {
        return 21;
    }
    else if(mode==22) {
        return 22;
    }
    else if(mode==23) {
        return 23;
    }
    else if(mode==24) {
        return 24;
    }
    else if(mode==25) {
        return 25;
    }
    else if(mode==26) {
        return 26;
    }
    else if(mode==27) {
        return 27;
    }
    else if(mode==28) {
        return 28;
    }
    else if(mode==29) {
        return 29;
    }
    else if(mode==30) {
        return 30;
    }
    else if(mode==31) {
        return 31;
    }
    else if(mode==32) {
        return 32;
    }
    else if(mode==33) {
        return 33;
    }
    else if(mode==34) {
        return 34;
    }
    else if(mode==35) {
        return 35;
    }
    else if(mode==36) {
        return 36;
    }
    else if(mode==37) {
        return 37;
    }
    else if(mode==38) {
        return 38;
    }
    else if(mode==39) {
        return 39;
    }
    else if(mode==40) {
        return 40;
    }
    else if(mode==41) {
        return 41;
    }
    else if(mode==42) {
        return 42;
    }
    else if(mode==43) {
        return 43;
    }
    else if(mode==44) {
        return 44;
    }
    else if(mode==45) {
        return 45;
    }
    else if(mode==46) {
        return 46;
    }
    else if(mode==47) {
        return 47;
    }
    else if(mode==48) {
        return 48;
    }
    else if(mode==49) {
        return 49;
    }
    else if(mode==50) {
        return 50;
    }
    else if(mode==51) {
        return 51;
    }
    else if(mode==52) {
        return 52;
    }
    else if(mode==53) {
        return 53;
    }
    else if(mode==54) {
        return 54;
    }
    else if(mode==55) {
        return 55;
    }
    else if(mode==56) {
        return 56;
    }
    else if(mode==57) {
        return 57;
    }
    else if(mode==58) {
        return 58;
    }
    else if(mode==59) {
        return 59;
    }
    else if(mode==60) {
        return 60;
    }
    else if(mode==61) {
        return 61;
    }
    else if(mode==62) {
        return 62;
    }
    else if(mode==63) {
        return 63;
    }
    else if(mode==64) {
        return 64;
    }
    else if(mode==65) {
        return 65;
    }
    else if(mode==66) {
        return 66;
    }
    else if(mode==67) {
        return 67;
    }
    else if(mode==68) {
        return 68;
    }
    else if(mode==69) {
        return 69;
    }
    else if(mode==70) {
        return 70;
    }
    else if(mode==71) {
        return 71;
    }
    else if(mode==72) {
        return 72;
    }
    else if(mode==73) {
        return 73;
    }
    else if(mode==74) {
        return 74;
    }
    else if(mode==75) {
        return 75;
    }
    else if(mode==76) {
        return 76;
    }
    else if(mode==77) {
        return 77;
    }
    else if(mode==78) {
        return 78;
    }
    else if(mode==79) {
        return 79;
    }
    else if(mode==80) {
        return 80;
    }
    else if(mode==81) {
        return 81;
    }
    else if(mode==82) {
        return 82;
    }
    else if(mode==83) {
        return 83;
    }
    else if(mode==84) {
        return 84;
    }
    else if(mode==85) {
        return 85;
    }
    else if(mode==86) {
        return 86;
    }
    else if(mode==87) {
        return 87;
    }
    else if(mode==88) {
        return 88;
    }
    else if(mode==89) {
        return 89;
    }
    else if(mode==90) {
        return 90;
    }
    else if(mode==91) {
        return 91;
    }
    else if(mode==92) {
        return 92;
    }
    else if(mode==93) {
        return 93;
    }
    else if(mode==94) {
        return 94;
    }
    else if(mode==95) {
        return 95;
    }
    else if(mode==96) {
        return 96;
    }
    else if(mode==97) {
        return 97;
    }
    else if(mode==98) {
        return 98;
    }
    else if(mode==99) {
        return 99;
    }
    else if(mode==100) {
        return 100;
    }
}; 

function testContexts() {
    var i;
    var j;
    var ITER = 1000;
    var CALLS = 100;
    var beginning_time = new Date().getTime();
    var foo = new Foo();
    for(i = 0; i < ITER; i++) {
        num = 1 + Math.floor(Math.random()*100);
        for(j=1; j<= CALLS; j++) {
            foo.test(num);
        }
    }
    var end_time = new Date().getTime();
    return end_time - beginning_time;
} 

function runs(run) {
    var fs = require("fs");
    var times;
    fs.writeFileSync("./runs_ifs/100/test"+run+".csv", "");
    fs.appendFileSync("./runs_ifs/100/test"+run+".csv", "iteration, Elapsed time in milliseconds \n");
    for(times = 1; times <= 100; times++) {
        var elapsed_time = testContexts();
        fs.appendFileSync("./runs_ifs/100/test"+run+".csv", times + "," + elapsed_time + "\n");
    }
}

var run;
for(run=1; run <=10; run++) {
    runs(run);
}

